### ISSUE TITLE
deposit_scale_factor in __init__ might cause Casper to stall

### ISSUE THREAD

**Issue by darcius** [ethereum/casper#106]
Found this while testing some code with Casper today when deploying to a testnet where the contract is deployed after the first epoch has passed. I might be missing something here, but it looks like this issue would stop Casper working immediately when deployed to any network that has been running for a while, including mainnet.

## The issue

An example where Casper is deployed to a network that is on it's `1000`th block and epoch length for Casper is `50` blocks. When initialised, Caspers `current_epoch` is set as `self.current_epoch = floor(block.number / self.EPOCH_LENGTH)`, so in this example, it would set its `current_epoch` as `20`.

Now this main issue is with `self.deposit_scale_factor` in the constructor. It's an array of deposit scale factors with epochs as keys. However it is initialised as `self.deposit_scale_factor[0] = 10000000000.0` which is what creates the issue as it assumes the current epoch is 0 and sets the scale factor for this.

https://github.com/ethereum/casper/blob/2f5ae45ad8e38943232330427eca3c75062bd564/casper/contracts/simple_casper.v.py#L143

When a deposit is attempted to be made in this scenario, it will attempt to calculate the scaled deposit for the new potential validator as such `scaled_deposit: decimal(wei/m) = msg.value / self.deposit_scale_factor[self.current_epoch]`. If our `current_epoch` is still `20`, the `self.deposit_scale_factor[self.current_epoch]` will return as 0 and cause the divide to throw, thus meaning no deposits can be made at all and Casper is stuck.

https://github.com/ethereum/casper/blob/2f5ae45ad8e38943232330427eca3c75062bd564/casper/contracts/simple_casper.v.py#L303

## The solution

A possible solution I've tested is to move `self.deposit_scale_factor[0] = 10000000000.0`  down two lines and change it too `self.deposit_scale_factor[self.current_epoch] = 10000000000.0` which fixed the issue for me.

**Comment by djrtwo** [comment]
Thank you! I'm on mobile, but after a quick look, I think you are correct.

Most of our testing thus far has been in deploying the Casper contract at
the genesis instead of mid chain. Because the new EIP dictates doing it mid
chain, it's time to enhance testing on that front.

Great find :)

**Comment by darcius** [comment]
Thanks @djrtwo! I'm currently implementing Casper directly into our unit tests which is how I found it. I'll let you guys know if I come across any other issues.

If that solution looks good to you, I can submit a PR, just let me know!

**Comment by djrtwo** [comment]
Yes! I think it would be difficult to test this in our current test setup, so can you just put a "TODO" comment above the `self.deposit_scale_factor[self.current_epoch] = 10000000000.0` line?

could be:
```python
# TODO: test deposit_scale_factor when deploying when current_epoch > 0
```

**Comment by djrtwo** [comment]
addressed in #107
